tripy/tripy/frontend/trace/ops/utils.py

```diff
@@ -42,29 +42,43 @@ def is_minus_one(arg):
     return isinstance(arg, int) and arg == -1
 
 
-# TODO (pranavm): Remove get_trace_shape
-
 ##
-## Inferring shape lengths (helpers)
+## Inferring shape helpers
 ##
 
 
-def get_trace_shape(input: "TraceTensor") -> Sequence[int]:
+def infer_broadcasted_shape(*input_shapes: Sequence[List[int]]):
     """
-    Given an operator input tensor, return its shape if it has already been given
-    or get its shape from the shape context if it's needed.
+    Given dynamic input shapes of trace tensors, infers a broadcasted shape.
+    This does not do any error checking since that can be done more reliably
+    later in the compiler.
     """
-    if input.shape is None:
-        from tripy.backend.mlir.utils import ShapeContext
+    max_rank = max(len(shape) for shape in input_shapes)
+    input_shapes = [[1] * (max_rank - len(shape)) + shape for shape in input_shapes]
+    return [max(dim) for dim in zip(*input_shapes)]
+
+
+class InferRankPolicies:
+    def same_as_input(idx=0):
+        def impl(self):
+            self.outputs[0].rank = self.inputs[idx].rank
+
+        return impl
+
+    def same_as_shape_of_shape_input(idx=0):
+        def impl(self):
+            # Expects that the first input is a shape tensor.
```

Comment:
```suggestion
            # Expects that the inputs[idx] is a shape tensor.
```

Reply:
Actually don't even need the comment since the assertion says the exact same thing. Will remove.